### DISCUSSION TITLE
BITS-Training mit "Zertifikat"?

### DISCUSSION THREAD

**Discussion by Hank0606** [BITS-Training/BITS-hugo#154] [Show and tell]
Gibt es Pläne – oder eine unaufwändige Möglichkeit – eine Art personalisiertes Abschlusszertifikat oder eine Teilnahmebescheinigung zu generieren, das auf den Quiz-/Testergebnissen basiert? In der Theorie stelle ich es mir so vor, dass der Prüfling (ich hoffe, es ist ok, wenn ich hier nicht gendere) zu Beginn seinen Vor- und Nachnamen angibt. Im Verlauf des Trainings werden die Quiz-Ergebnisse gespeichert und am Ende ein PDF generiert, dass neben der formellen Teilnahme-/Prüfbestätigung die Werte aus dem Quiz und den Namen beinhaltet.

**Comment by BITS-Editor** [comment]
Danke für die Anregung. Das Thema kommt immer wieder - zu recht - auf.

Leider haben wir noch keine stabile Idee die ohne Datenbank auskommt entwickelt.

Cookies, die man selbst füllt, wären ein Ansatz, aber alles andere als manipulationssicher.

**Comment by andi-blafasl** [comment]
Da das BITS komplett ohne Backend-System arbeitet, aus Datenschutzgründen keine Speicherung von personenbezogenen Daten stattfindet und keine Cookies gesetzt werden, ist es aktuell nicht möglich hier etwas zu realisieren.
Denkbar wäre meiner Meinung nach eine Lösung über ein Lösungswort im Quiz. Das Kann man dann abfragen und hat den Nachweis das zumindest das Quiz einmal mit den richtigen antworten abgelegt wurde. Solch ein "Nachweissystem" kann jeder Admin in seiner Firma selbst umsetzten.

**Comment by Hank0606** [comment]
Vielen Dank für die Info, das hatte ich auch als einen ersten Ansatz in Erwägung gezogen. – Da meine technischen Kenntnisse begrenzt sind, die Frage, wie denn der Haken für die abgearbeiteten Lektionen "am Leben erhalten" wird; läuft das nicht über Cookies?

**Comment by andi-blafasl** [comment]
Der Verlauf wird im Session-Store des Browsers abgelegt. Kein Cookie nötig.

**Comment by BITS-Editor** [comment]
Wäre vielleicht der LocalStorage eine Betrachtung wert? Verbunden mit einem Button, mit dem man diesen dann selbst leeren kann? https://www.tutorialrepublic.com/html-tutorial/html5-web-storage.php

**Comment by andi-blafasl** [comment]
Alles was Local im Browser gespeichert wird, macht für mich keinen sinn. Das kann man zu leicht manipulieren und ohne zentrale Verwaltung der Teilnehmer und Überprüfung der Lernziele hat ein Zertifikat doch auch keine Aussagekraft.

**Comment by BITS-Editor** [comment]
Das stimmt. Das wäre auch nur eine absolute Schmalspurlösung. Könntest Du den Aufwand für einen Test abschätzen?

**Comment by andi-blafasl** [comment]
Puh, da ich davon so gut wie keine Ahnung habe, müsste ich mich erst mal ziemlich in die Materie einarbeiten. Bin ja kein Programmierer und da braucht man schon ein bisschen JavaScript für, zumindest so weit ich das auf die schnelle Verstanden habe. Und damit würden wir ja wieder Personenbezogenen Daten Speichern und sowas wie Cookies dauerhaft beim Benutzer ablegen. Das muss ja dann auch wieder irgendwie in die Datenschutzerklärung und eventuell sogar mit Opt-In Option oder???

**Comment by BITS-Editor** [comment]
Die Datenschutzerklärung ist ja fix gemacht, da sehe ich nicht das Problem.

Die Technik ist mE das größere Problem. Ich schlage vor, wir lassen das Issue einfach offen. Vielleicht fällt uns oder jemand anderen noch etwas ein.

**Comment by BITS-Editor** [comment]
Ich denke, man sollte vielleicht doch einmal das JS-Feature localStorage für die Namenseingabe testen. https://blog.logrocket.com/localstorage-javascript-complete-guide/
Eine Demo gibt es hier:
http://jsfiddle.net/Benjol/HMEVd/
Beispiele gibt es hier:
https://css-tricks.com/localstorage-examples/ - inkl. Benutzernamen-Speicherung
Mir ist bewusst, dass das alles andere als sicher und verlässlich ist, aber die Zahl der Anfragen nach Nutzer-Zertifikaten steigt bei mir zusehends.
Da die Inhalte im localStorage dauerhaft gespeichert werden, müssten sie irgendwie überschrieben werden können.

**Comment by BITS-Editor** [comment]
Weitere Möglichkeiten nach https://blog.logrocket.com/beyond-cookies-todays-options-for-client-side-data-storage/ wären
- sessionStorage
- [localForage](https://github.com/localForage/localForage) (mit F)
- client side cookies (also im Client generiert - hier wäre die Frage, woher der Value eingefügt wird; am besten wohl über ein Feld, das man selbst befüllen muss). Vorteil: man kann hier expiration date setzen Beispiel: https://www.geeksforgeeks.org/how-to-create-and-read-value-from-cookie/
- IndexedDB

**Comment by BITS-Editor** [comment]
@andi-blafasl prüft eine ganz neue Idee, die halbautomatisiert funktionieren könnte.

**Comment by BITS-Editor** [comment]
Bei LocalStorage kann nur die ursprüngliche Domain die Daten auslesen, so dass eine gewisse Sicherheit gegenüber anderen Webseiten gegeben ist - aber natürlich nicht gegenüber User-Manipulationen. Das finde ich aber nicht so schlimm.

Beim Absenden könnte dann die DeleteItem- oder Clean-Funktion mit ausgelöst werden, um die Daten zu löschen.

**Comment by BITS-Editor** [comment]
Eine beispielhafte Umsetzung mit LocalStorage als Speicherort im Browser findet sich hier:
https://github.com/BITS-Training/Userzertifikat
